.github/workflows/build_and_release_gem.yml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2021 SolarWinds, LLC.
+# All rights reserved.
+
+name: BUILD, VERIFY, and RELEASE Ruby Gem to RubyGems
+
+# This workflow relies on the following secrets
+# - TRACE_BUILD_RUBY_ACTIONS_API_TOKEN (from the trace-build user in github)
+# - RUBYGEMS_TOKEN (from the appoptics user on rubygems.org)
+# - APPOPTICS_SERVICE_KEY
+# - APPOPTICS_COLLECTOR
+#
+# The correct oboe version has to be in the file ext/oboe_metal/src/VERSION
+
+on:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build + Release Gem to RubyGems
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Ruby 2.6 and bundle
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.6
+
+      - name: Install gems
+        run: |
+          echo 'gem: --no-document' >> ~/.gemrc
+          bundle install --without development --without test
+
+      - name: Install swig 4.0.2
+        run: |
+          curl -SL http://kent.dl.sourceforge.net/project/swig/swig/swig-4.0.2/swig-4.0.2.tar.gz | tar xzC /tmp
+          cd /tmp/swig-4.0.2
+          ./configure && make && sudo make install
+          cd -
+
+      - name: Download oboe files from github
+        env:
+          TRACE_BUILD_TOKEN: ${{ secrets.TRACE_BUILD_RUBY_ACTIONS_API_TOKEN }}
+        run: |
+          oboe_version=`cat ext/oboe_metal/src/VERSION`
+          bundle exec rake oboe_github_fetch[$oboe_version]
+          ls -l ext/oboe_metal/verify
+          ls -l ext/oboe_metal/verify/bson
+
+      - name: Download files from files.appoptics.com and create swig wrapper
+        run: |
+          oboe_version=`cat ext/oboe_metal/src/VERSION`
+          bundle exec rake oboe_files_appoptics_fetch[$oboe_version]
+          bundle exec rake oboe_verify
+
+      - name: Build Gem and upload to Rubygems
+        id: build
+        env:
+          RUBYGEMS_TOKEN: ${{ secrets.RUBYGEMS_TOKEN }}
+        run:  |
+          bundle exec rake build_gem
+          checksum=`shasum -a256 *.gem | awk '{print $1;}'`
+          echo "checksum: $checksum"
+          echo ::set-output name=checksum::$checksum
+          echo ::set-output name=gem_version::`ruby -e 'require "./lib/appoptics_apm/version"; puts AppOpticsAPM::Version::STRING'`
```

Comment:
In the workflow run for this step seeing the following warning:
> WARNING:  open-ended dependency on json (>= 0) is not recommended
  use a bounded requirement, such as '~> x.y'
Anything to be concerned with?

Reply:
no concern, I'll put something back

Reply:
Looks like this warning is caused by https://github.com/appoptics/appoptics-apm-ruby/blob/master/appoptics_apm.gemspec#L46

Is it really not a concern? [Rubygems](https://github.com/rubygems/rubygems/blob/d8d235f1eeb6da0fb289ce0415ad049153f9adc9/lib/rubygems/specification.rb#L2897) seems to want you not to do that.